pkg/agent/cniserver/pod_configuration_windows.go

```diff
@@ -0,0 +1,113 @@
+// in another goroutine. The function is for ContainerD runtime. The host interface is created after
+// CNI call completes.
+func (pc *podConfigurator) connectInterfaceToOVSAsync(ifConfig *interfacestore.InterfaceConfig, containerAccess *containerAccessArbitrator) error {
+	if containerAccess == nil {
+		return fmt.Errorf("container lock cannot be null")
+	}
+	ovsPortName := ifConfig.InterfaceName
+	expectedEp, ok := pc.ifConfigurator.getEndpoint(ovsPortName)
+	if !ok {
+		return fmt.Errorf("failed to find HNSEndpoint %s", ovsPortName)
+	}
+	hostIfAlias := fmt.Sprintf("%s (%s)", util.ContainerVNICPrefix, ovsPortName)
+	containerID := ifConfig.ContainerID
+	go func() {
+		klog.Infof("Waiting for interface %s to be created", hostIfAlias)
+		err := wait.PollImmediate(time.Second, 60*time.Second, func() (bool, error) {
+			curEp, ok := pc.ifConfigurator.getEndpoint(ovsPortName)
+			if !ok {
+				return true, fmt.Errorf("failed to find HNSEndpoint %s", ovsPortName)
+			}
+			if curEp.Id != expectedEp.Id {
+				klog.Warningf("Detected HNSEndpoint change for port %s, exiting current goroutine", ovsPortName)
+				return true, nil
+			}
+			if !util.HostInterfaceExists(hostIfAlias) {
+				klog.Infof("Waiting for interface %s to be created", hostIfAlias)
+				return false, nil
+			}
+			if err := pc.connectInterfaceToOVSCommon(ovsPortName, ifConfig); err != nil {
+				return true, fmt.Errorf("failed to connect to OVS for container %s: %v", containerID, err)
+			}
+			return true, nil
+		})
+		if err != nil {
+			klog.Errorf("Failed to create OVS port for container %s: %v", containerID, err)
+		}
+	}()
+	return nil
+}
+
+// connectInterfaceToOVS connects an existing interface to OVS br-int.
+func (pc *podConfigurator) connectInterfaceToOVS(
+	podName string,
+	podNameSpace string,
+	containerID string,
+	hostIface *current.Interface,
+	containerIface *current.Interface,
+	ips []*current.IPConfig,
+	containerAccess *containerAccessArbitrator,
+) (*interfacestore.InterfaceConfig, error) {
+	// Use the outer veth interface name as the OVS port name.
+	ovsPortName := hostIface.Name
+	containerConfig := buildContainerConfig(ovsPortName, containerID, podName, podNameSpace, containerIface, ips)
+	hostIfAlias := fmt.Sprintf("%s (%s)", util.ContainerVNICPrefix, ovsPortName)
+	// - For ContainerD runtime, the container interface is created after CNI replying the network setup result.
+	//   So for such case we need to use asynchronous way to wait for interface to be created.
+	// - For Docker runtime, antrea-agent still creates OVS port synchronously.
+	// - Here antrea-agent determines the way of OVS port creation by checking if container interface is yet created.
+	//   If one day ContainerD runtime changes the behavior and container interface can be created when attaching
+	//   HNSEndpoint/HostComputeEndpoint. The current implementation sill works. It will choose the synchronized
```

Comment:
this should be a single sentence, not two: "If one day ContainerD runtime changes the behavior and container interface can be created when attaching HNSEndpoint/HostComputeEndpoint, the current implementation will still work."

Reply:
Done.